dist/api_data.py

```diff
@@ -297,14 +297,21 @@ def __ge__(self, other):
         WT_CONNECTION::add_compressor.  If WiredTiger has builtin support for
         \c "lz4", \c "snappy", \c "zlib" or \c "zstd" compression, these names
         are also available.  See @ref compression for more information'''),
-    Config('checksum', 'uncompressed', r'''
-        configure block checksums; permitted values are <code>on</code>
-        (checksum all blocks), <code>off</code> (checksum no blocks) and
-        <code>uncompresssed</code> (checksum only blocks which are not
-        compressed for any reason).  The \c uncompressed setting is for
-        applications which can rely on decompression to fail if a block
-        has been corrupted''',
-        choices=['on', 'off', 'uncompressed']),
+    Config('checksum', 'on', r'''
+        configure block checksums; the permitted values are \c on, \c off, \c uncompressed and
+        \c unencrypted. The default is \c on, in which case all block writes include a checksum
+        subsequently verified when the block is read. The \c off setting does no checksums,
+        intended for applications with storage systems that detect block corruption. The
+        \c uncompressed setting only checksums blocks that are not compressed; the \c unencrypted
+        setting only checksums blocks that are not encrypted. The \c uncompressed and
+        \c unencrypted settings are intended for applications with compression or encryption
+        engines that detect block corruption. Corruption detection is often the case for
+        encryption engines, including the \c sodium engine included in WiredTiger. Corruption
+        detection is rarely the case for compression engines, specifically, none of the \c lz4,
+        \c snappy, \c zlib and \c zstd compression engines included in WiredTiger have checksum
```

Comment:
Do we want to itemize these here? If/when the list changes someone will have to remember and find this place to change. I won't push hard as it is useful as written, only considering future maintenance.

Reply:
I think I'm missing your point, I don't see how itemization makes the problem better?

Reply:
I meant itemizing makes maintenance harder because if we stop including, say, snappy then someone has to remember to fix the checksum description which is not an obvious connection. Hopefully they'd grep.

Reply:
That paragraph only lives in the tuning doc, now, so I think it's likely to be found.